build_tools/bazel_to_cmake/bazel_to_cmake.py

```diff
@@ -103,11 +103,12 @@ def convert_directory(directory_path, write_files, allow_partial_conversion):
   if not os.path.isdir(directory_path):
     raise FileNotFoundError(f"Cannot find directory '{directory_path}'")
 
+  skip_file_path = os.path.join(directory_path, ".skip_bazel_to_cmake")
   build_file_path = os.path.join(directory_path, "BUILD")
   cmakelists_file_path = os.path.join(directory_path, "CMakeLists.txt")
 
-  if not os.path.isfile(build_file_path):
-    # No Bazel BUILD file in this directory to convert, skip.
+  if os.path.isfile(skip_file_path) or not os.path.isfile(build_file_path):
+    # No Bazel BUILD file in this directory or explicit skip.
```

Comment:
This was already merged when I spotted this (sorry for the delay), but we've already got a pattern `EDIT_BLOCKING_PATTERN` you can put in the CMakeLists.txt file which will make the script fall back to printing what it would edit to stdout. Should we consolidate these two skipping methods? The idea behind falling back to stdout was that for something that wasn't fully automated, you might still want to see what the script would generate. It seems like here you want to always skip entirely.

Some options:
1. Drop `EDIT_BLOCKING_PATTERN` and just use this
2. Drop this and just use `EDIT_BLOCKING_PATTERN`
3. (2), but also make `EDIT_BLOCKING_PATTERN` not do the stdout thing. Just skip the file
4. (2), but also make `EDIT_BLOCKING_PATTERN` only apply in strict mode and otherwise write the file anyway (basically `EDIT_BLOCKING_PATTERN` becomes a manual way to tell bazel_to_cmake that it can't write this file cleanly.
5. (2), but add an additional pattern to tell bazel_to_cmake to skip a file and not print to stdout.

Personally, I like option 4. I think it would make the behavior more consistent and still allows someone to see what bazel_to_cmake would do for any given file without spamming the logs by default.

Reply:
I don't think I have much of an opinion as long as we have an obvious way to tell it that a directory is opted out (which, if I read properly, can be a comment in the CMakeLists file itself. Can you point exactly at what would need to change (or feel free to edit -- I'm not married to how I did this)?

Reply:
You just put some comment in the CMakeLists.txt file like `bazel_to_cmake: do not edit`

The actual regex is

```py
EDIT_BLOCKING_PATTERN = re.compile(
    r"bazel[\s_]*to[\s_]*cmake[\s_]*:?[\s_]*do[\s_]*not[\s_]*edit",
    flags=re.IGNORECASE)
```